plugins/telescope/default.nix

```diff
@@ -74,16 +89,28 @@ in {
           defaults = cfg.defaults;
         }
         // cfg.extraOptions;
-    in ''
-      do
-        local __telescopeExtensions = ${helpers.toLuaObject cfg.enabledExtensions}
+    in
+      ''
+        do
+          local __telescopeExtensions = ${helpers.toLuaObject cfg.enabledExtensions}
 
-        require('telescope').setup(${helpers.toLuaObject options})
+          require('telescope').setup(${helpers.toLuaObject options})
 
-        for i, extension in ipairs(__telescopeExtensions) do
-          require('telescope').load_extension(extension)
+          for i, extension in ipairs(__telescopeExtensions) do
+            require('telescope').load_extension(extension)
+          end
         end
-      end
-    '';
+      ''
+      + (optionalString (cfg.keymaps != {}) ''
```

Comment:
Instead of adding keymaps here would you not be able to do something like `config.maps.normal = mapAttrs (<some function to build the correct keymaps>) cfg.keymaps` ?

I think this would avoid repetition with the keymaps, and it will benefit from any change we do to the keymaps.

Reply:
Yes, I thought about it, but I wanted to `require("telescope.builtin")` only once.

Reply:
Yes, I will wait for the merge of https://github.com/pta2002/nixvim/pull/178 and will use the nixvim way then.